mantle/util/repo.go

```diff
@@ -82,6 +83,27 @@ func GetLatestLocalBuild(root, arch string) (*LocalBuild, error) {
 	return GetLocalBuild(root, "latest", arch)
 }
 
+func GetRelativeLocalBuildId(root, relid string) (string, error) {
+	// resolve relative builds, e.g. -1, -2
+	relBuild, err := strconv.ParseInt(relid, 10, 0)
+	if err != nil {
+		return "", fmt.Errorf("invalid relative build index %s: %w", relid, err)
+	}
+	if relBuild >= 0 {
+		return "", fmt.Errorf("invalid relative build index %s: not negative", relid)
+	}
+	// we know it'll fit in an int since we used bitSize: 0 above
+	relBuildInt := int(-relBuild)
+	builds, err := cosa.GetBuilds(filepath.Join(root, "builds"))
+	if err != nil {
+		return "", err
+	}
+	if relBuildInt >= len(builds.Builds) {
+		return "", fmt.Errorf("relative build index %d does not exist", -relBuild)
+	}
+	return builds.Builds[relBuildInt].ID, nil
```

Comment:
I have not checked here, but is the build in position 0 in the array the most recent one or the oldest one?

If it's the later then don't we need something like `size - id` instead?

Reply:
It's the most recent one.